app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/utils/SheetsManager.java

```diff
@@ -128,18 +149,151 @@ public boolean verifySheet(Context context, String sheetID) {
             List<Object> parentObject = fetchData(context, sheetID, "BackendData!2:2").get(0);
             String version = parentObject.get(parentObject.size() - 1).toString();
             return isSheetVersionSupported(version);
-        } catch (Exception e){
+        } catch (Exception e) {
             return false;
         }
     }
 
     private boolean isSheetVersionSupported(String version) {
 
         List<String> supportedVersion = new ArrayList<>();
-        supportedVersion.add("3.1.0");
-        supportedVersion.add("3.0");
-        supportedVersion.add("2.8");
+        supportedVersion.add(threeOne);
+        supportedVersion.add(three);
+        supportedVersion.add(twoEight);
+
+        sessionConfig.setSheetVersion(version);
 
         return supportedVersion.contains(version);
     }
+
+    private void getTransactionCategories(Context context, String sheetID) {
+        String range = "";
+
+        switch (sessionConfig.getSheetVersion()) {
+            case twoEight:
+                range = "BackendData!B2:B";
+                break;
+            case three:
+            case threeOne:
+                range = "BackendData!F2:F";
+                break;
+            case threeTwo:
+                range = "BackendData!G2:G";
+                break;
+        }
+
+        List<List<Object>> data = fetchData(context, sheetID, range);
+        transactionCategories = convertToOneDimension(data);
+    }
+
+    private void getTransactionAccounts(Context context, String sheetID) {
+        String range = "";
+
+        switch (sessionConfig.getSheetVersion()) {
+            case twoEight:
+                range = "BackendData!E2:E";
+                break;
+            case three:
+                range = "BackendData!H2:H";
+                break;
+            case threeOne:
+                range = "BackendData!J2:J";
+                break;
+            case threeTwo:
+                range = "BackendData!M2:M";
+                break;
+        }
+
+        List<List<Object>> data = fetchData(context, sheetID, range);
+        transactionAccounts = convertToOneDimension(data);
+    }
+
+    private List<String> convertToOneDimension(List<List<Object>> data) {
+        List<String> result = new ArrayList<>();
```

Comment:
Aren't there `filter` and `map` operations in Android/Java?

Reply:
I tried finding but couldnt find any, or I am not aware of the method so I used this one.